lnwallet/channel.go

```diff
@@ -3591,6 +3760,14 @@ func ChanSyncMsg(c *channeldb.OpenChannel,
 	// allowing us to sweep our funds.
 	if isRestoredChan {
 		currentCommitSecret[0] ^= 1
+
+		// If this is a tweakless channel, then we'll purposefully send
+		// a next local height thta's invalid to trigger a force close
+		// on their end. We do this as tweakless channels don't require
+		// that the commitment point is valid, only that it's present.
+		if c.ChanType.IsTweakless() {
+			nextLocalCommitHeight--
```

Comment:
This probably works, but I find it a bit hard to analyze. Won't this result in a local commit height equal to the situation where the last ACK we sent got lost (and hence we increased the height, but the remote did not)?

I'm also not sure what happens in the zero-state case.

Could a more bulletproof solution be to set it to `0` always? That would still leave the zero-state case an edge case.

The best solution would be to change the spec to always send the correct commitpoint (as before), but that train has probably left...

Reply:
Modified to instead always set to zero.